linera-chain/src/manager/multi.rs

```diff
@@ -16,16 +16,23 @@ use linera_base::{
     identifiers::{ChainId, Owner},
 };
 use linera_execution::committee::Epoch;
+use rand_chacha::{rand_core::SeedableRng, ChaCha8Rng};
+use rand_distr::{Distribution, WeightedAliasIndex};
 use serde::{Deserialize, Serialize};
-use std::collections::HashMap;
+use std::collections::{BTreeMap, HashMap};
 use tracing::error;
 
 /// The specific state of a chain managed by multiple owners.
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct MultiOwnerManager {
-    /// The public keys of the chain's co-owners.
-    pub public_keys: HashMap<Owner, PublicKey>,
-    /// Latest authenticated block that we have received (and voted to validate).
+    /// The public keys of the chain's co-owners, with their weights. If all weights are zero, every owner can
+    /// propose in every round. Otherwise every round has only one leader, and rounds can time out.
+    pub public_keys: BTreeMap<Owner, (PublicKey, u128)>,
```

Comment:
No, we could make it smaller. Probably `u32` or `u16` is easily good enough in practice?

It's just that we'll have to round the stake then: Since it's represented by a `u128`, we'd have to divide it by some factor that makes the best use of the weight's type's precision without exceeding its maximum value.

Happy to change it. `u32`?

Reply:
`u64` probably just fine as well